### ISSUE TITLE
Unhanded exception: Sort output by missing column 

### ISSUE THREAD

**Issue by sambles** [OasisLMF/OasisLMF#452]
* `accded2other` not in accounts file (Not required)
* set `"oed_fields": ["accded2other"],` in analysis_settings.json

**Trace**
```
Traceback (most recent call last):
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2897, in get_loc
    return self._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 107, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 131, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1607, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1614, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'accded2other'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sam/repos/env/tpy/bin/oasislmf", line 7, in <module>
    sys.exit(RootCmd().run())
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/cli/root.py", line 41, in run
    return super(OasisBaseCommand, self).run(args=args)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/argparsetree/cmd.py", line 159, in run
    return cmd_cls(sub_command_name).run(args)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/argparsetree/cmd.py", line 159, in run
    return cmd_cls(sub_command_name).run(args)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/argparsetree/cmd.py", line 161, in run
    return self.action(args) or 0
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/cli/model.py", line 477, in action
    cmd.action(args)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/cli/model.py", line 371, in action
    user_data_dir=user_data_dir
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/utils/log.py", line 106, in wrapper
    result = func(*args, **kwargs)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/manager.py", line 662, in generate_model_losses
    ri=ri)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/utils/log.py", line 106, in wrapper
    result = func(*args, **kwargs)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/model_preparation/summaries.py", line 581, in generate_summaryxref_files
    'il'
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/utils/log.py", line 106, in wrapper
    result = func(*args, **kwargs)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/model_preparation/summaries.py", line 463, in get_summary_xref_df
    ) = group_by_oed(cols_group_by, map_df, exposure_df, id_set_index, accounts_df)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/model_preparation/summaries.py", line 191, in group_by_oed
    summary_group_df.fillna(0, inplace=True)
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/utils/data.py", line 164, in factorize_dataframe
    by_col_indices = by_col_indices or (None if not by_col_labels else [df.columns.get_loc(label) for label in by_col_labels])
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/oasislmf/utils/data.py", line 164, in <listcomp>
    by_col_indices = by_col_indices or (None if not by_col_labels else [df.columns.get_loc(label) for label in by_col_labels])
  File "/home/sam/repos/env/tpy/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2899, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 107, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 131, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1607, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1614, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'accded2other'

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@python.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    %config Application.verbose_crash=True
```

**Comment by sambles** [comment]
Warn user and default to all_risks? 
https://github.com/OasisLMF/OasisLMF/blob/develop/oasislmf/model_preparation/summaries.py#L169-L194

**Comment by sambles** [comment]
Fix - Report data issue and halt:  #457

**Comment by sambles** [comment]
#457 457
